app_package/src/showroom.ts

```diff
@@ -0,0 +1,328 @@
+                this._camera.animateToMatchmoveState(this._levitateState);
+                break;
+            case ShowroomState.Configure:
+                this._watch.setState(WatchState.Configure);
+                this._camera.animateToArcRotateState(this._configureState).then(() => {
+                    return Tools.DelayAsync(300);
+                }).then(() => {
+                    this._stateSetToConfigurationObservable.notifyObservers();
+                });
+                break;
+        }
+    }
+
+    public set showStuds(visibility: boolean) {
+        this._studs?.Mesh?.setEnabled(visibility);
+    }
+
+    private _configurationOptionsLoaded: boolean;
+    public get configurationOptionsLoaded(): boolean {
+        return this._configurationOptionsLoaded;
+    }
+    public configurationOptionsLoadedObservable: Observable<Showroom>;
+
+    public setZoomPercent(zoomPercent: number): void {
+        this._camera.arcRotateZoomPercent = zoomPercent;
+    }
+
+    public disableMouseWheel(): void {
+        this._camera.enableMouseWheel = false;
+    }
+
+    private constructor(scene: Scene, watch: Watch, params: IVaporwearExperienceParams) {
+        this._scene = scene;
+        this._state = ShowroomState.Overall;
+        this._stateSetToConfigurationObservable = new Observable();
+
+        this._watch = watch;
+
+        this._camera = new ShowroomCamera(this._scene);
+        this._camera.fov = 0.5;
+        this._camera.minZ = 0.01;
+        this._camera.maxZ = 100;
+        
+        this._overallState = {
+            matchmoveTarget: this._watch.cameraParentOverall,
+            focusDepth: this._watch.cameraParentOverall.position.length()
+        };
+        this._claspState = {
+            matchmoveTarget: this._watch.cameraParentClasp,
+            focusDepth: this._watch.cameraParentClasp.position.length()
+        };
+        this._faceState = {
+            matchmoveTarget: this._watch.cameraParentFace,
+            focusDepth: this._watch.cameraParentFace.position.length()
+        };
+        this._levitateState = {
+            matchmoveTarget: this._watch.cameraParentLevitate,
+            focusDepth: this._watch.cameraParentLevitate.position.length()
+        };
+        this._configureState = {
+            startingPosition: this._watch.cameraParentOverall.absolutePosition.clone(),
+            target: (this._watch.cameraParentOverall.parent! as TransformNode).absolutePosition.clone(),
+            lowerRadiusLimit: 3,
+            upperRadiusLimit: 15
+        };
+
+        this._watch.setState(WatchState.Overall);
+        this._camera.setToMatchmoveState(this._overallState);
+        this._state = ShowroomState.Overall;
+
+        const configurationPromise = new Promise<void>((resolve) => 
+        {
+            this._stateSetToConfigurationObservable.addOnce(() => {
+                resolve();
+            });
+        });
+
+        fetch(params.assetUrlRoot + "/" + params.assetUrlWatchStuds);
```

Comment:
I would suggest using string interpolation for these.

```suggestion
        fetch(`${params.assetUrlRoot}/${params.assetUrlWatchStuds}`);
```